inlang/source-code/sdk/src/createNewProject.ts

```diff
@@ -0,0 +1,31 @@
+import type { Repository } from "@lix-js/client"
+import { ProjectSettings } from "@inlang/project-settings"
+import { assertValidProjectPath, pathExists } from "./validateProjectPath.js"
+import { defaultProjectSettings } from "./defaultProjectSettings.js"
+
+/**
+ * Creates a new project in the given directory.
+ * The directory must be an absolute path, must not exist, and must end with {name}.inlang
+ * Uses defaultProjectSettings unless projectSettings are provided.
+ *
+ * @param projectPath - Absolute path to the [name].inlang directory
+ * @param repo - An instance of a lix repo as returned by `openRepository`
+ * @param projectSettings - Optional project settings to use for the new project.
+ */
+export async function createNewProject(args: {
+	projectPath: string
+	repo: Repository
+	projectSettings?: ProjectSettings
```

Comment:
I would suggest we make projectSettings non optional and export the default project settings for the consumer instead - In most cases the configuration will be allowed to be changed from default to specific. For example the language tags are most likely changed in all situations, Manage might allow you to choose the lints (with good defaults so) - To be able to change them one needs to access the defaults to make them visible - and the can pass them as an argument than.

Reply:
> I would suggest we make projectSettings non optional 
 What's the benefit of making it non-optional - i can see use cases where no interaction before creating a project is desireable. 
 
 > export the default project settings for the consumer instead
 are you suggesting that we expose defaultProjectSettings on the sdk api,

Reply:
Default can be forgoton/ignored. If the default case is that the consumer of the api will pass a settings object i would enforce the consumer to pass it at compile time. Documentation of the function can guide how to fill it with a default value - but it is an explicit decision instead of an optional choice.

Reply:
👍 
"explicit" is more readable too.